x-pack/plugins/security_solution/public/common/components/event_details/event_fields_browser.tsx

```diff
@@ -166,10 +169,22 @@ const useFieldBrowserPagination = () => {
 
 /** Renders a table view or JSON view of the `ECS` `data` */
 export const EventFieldsBrowser = React.memo<Props>(
-  ({ browserFields, data, eventId, isDraggable, timelineTabType, timelineId, isReadOnly }) => {
+  ({ browserFields, data, eventId, isDraggable, timelineTabType, scopeId, isReadOnly }) => {
     const containerElement = useRef<HTMLDivElement | null>(null);
     const dispatch = useDispatch();
-    const getTimeline = useMemo(() => timelineSelectors.getTimelineByIdSelector(), []);
+    const getScope = useMemo(() => {
+      if (isTimelineScope(scopeId)) {
```

Comment:
👌🏾

Reply:
I know there are anti-redux and likewise anti-context opinions, but I wonder if it's worth putting these items in a top level context of some sort since everyone doesn't use redux? I'm just imagining some future scope that we don't even consider and there being one more special use cases such as `isInFlyoutScope` or `isInSomePageScope`.

Reply:
Absolutely agree, this logic shouldn't be a part of the react functional component (I prefer to keep it stateless) and I'm planning to change this in the follow up PRs.